src/baseball_stats_python/utils/utils.py

```diff
@@ -0,0 +1,15 @@
+from datetime import datetime
+
+
+def validate_date_range(start_dt, end_dt):
+    date_format = "%Y-%m-%d"
+    
+    try:
```

Comment:
I would prefer `try-except` format first then check range
For example:
```python
try:
    start_date = datetime.strptime(start_dt, date_format)
    end_date = datetime.strptime(end_dt, date_format)
except ValueError as e:
    raise ValueError(e)

if end_date < start_date:
    raise ValueError("end_dt cannot be earlier than start_dt.")
```

Thus, we can know the error is from date format or date range.

Reply:
Also `start_dt` and `end_dt` are allowed to have one empty string `""` between them.

Ex: `{"start_dt": "", "end_dt": "2024-10-27"}` or `{"start_dt": "2024-10-01", "end_dt": ""}`

We might also want to handle these situations